File: problems/5e397e4f-e262-498c-a955-c30aa0900d52/ruby/checker.rb
require_relative 'backend_checker'
require_relative 'solution'

def run_checker(testcases)
  run_result = BackendChecker::RunResult.new
  
  testcases.each do |testcase|
    input = testcase[:input]
    expected = testcase[:expected]
    
    capture = BackendChecker.capture_stdout { roman_to_integer(input) }
    got = capture[:result]
    
    if expected != got
      failed_testcase = BackendChecker::FailedTestcase.new(
        testcase[:id],
        testcase[:input],
        testcase[:expected],
        got,
        capture[:stdout]
      )
      
      run_result.failedTestcases << failed_testcase
      run_result.isPassed = false
    end
  end
  
  run_result
end

File: problems/5e397e4f-e262-498c-a955-c30aa0900d52/cpp/checker.cpp
#include "backend_checker.cpp"
#include "driver.cpp"
#include "solution.cpp"

namespace MainChecker {
    BackendChecker::RunResult runChecker(const std::vector<Driver::Testcase>& testcases) {
        BackendChecker::RunResult runResult;
        Solution solution;

        for (const auto& testcase : testcases) {
            std::string input = testcase.input;
            int expected = testcase.expected;
            
            auto capture = BackendChecker::captureStdout([&]() {
                return solution.romanToInteger(input);
            });
            
            int got = capture.result;

            if (expected != got) {
                BackendChecker::FailedTestcase failedTestcase;
                failedTestcase.id = testcase.id;
                failedTestcase.input = testcase.input;
                failedTestcase.expected = testcase.expected;
                failedTestcase.got = got;
                failedTestcase.stdOut = capture.stdout;

                runResult.failedTestcases.push_back(failedTestcase);
                runResult.isPassed = false;
            }
        }

        return runResult;
    }
}
